org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeCoordinatorHandler.java

```diff
@@ -75,7 +77,7 @@
  * @author Chris Jackson - Initial contribution
  */
 public abstract class ZigBeeCoordinatorHandler extends BaseBridgeHandler
```

Comment:
You should imho extract an interface `ZigBeeCoordinator`, which is then used in the discovery - this would decouple the discovery infrastructure from thing handlers from an API perspective. See also the `BluetoothAdapter` interface as a reference, where it is done that way.

Reply:
Yes, you are probably correct, but again I'd prefer not to link that to this PR if possible?

Reply:
Hm, that's one of the central pieces of the whole discovery participant setup, so I would clearly see it as a part of this PR. Can you explain me where you see the issue with simply extracting a new interface? It should imho not really have much impact on any of the existing code apart from the new discovery service.

Reply:
I don't really see why it's linked. The coordinator handler was there before and the API hasn't changed. I want to avoid refactoring this now, and then doing it again in a month or two to completely change the naming etc.

Reply:
>If there's a bigger refactoring in the Zigbee thing handlers soon, this would rather be another reason for me to introduce this interface now

Ok, so you want to wrap up future refactoring into this PR? At the moment I don't have this finished, but if we want to wrap this all up into the single PR, it will take some time and I was really hoping to get this PR finished sooner than that.

Reply:
> so you want to wrap up future refactoring into this PR?

No, not at all! I have no idea what this future refactoring will be about, but it should clearly not part of this PR. I was only talking about introducing the dedicated interface that is used by the discovery service and which will NOT have to be a part of any future refactoring (as I very much hope that you do not plan to do a bigger refactoring to the discovery service, which we were just settling here).

Reply:
Ok, I thought you wanted the whole coordinator API specified - if it's just the methods that are used by discovery, then I can make an interface with the 3 or 4 methods in it, that's fine. These methods are part of a wider interface though, and I assumed that you wanted that interface specified?

Reply:
No, it is really just about the interface that is to be used [here](https://github.com/openhab/org.openhab.binding.zigbee/pull/169#discussion_r209867328) and registered as a service [here](https://github.com/openhab/org.openhab.binding.zigbee/pull/169#discussion_r209869035). This interface should only define what is relevant for discovery, nothing else. You can refactor the implementations of coordinator handlers in any way you like afterwards and also introduce additional interfaces, if desired. Just the discovery interface (if ZigBeeCoordinator isn't a good name because it might be misleading, I am fine with any other choice) should then be kept as is.

Reply:
Ok, that's fine, since you aren't after the coordinator interface, I will create a new interface just for the discovery methods.

Reply:
@kaikreuzer having looked at this some more, I would prefer to merge this now, and look at updating the interface in a few weeks. The issue is that as mentioned above, I want to refactor this soon, and the reason is to add support both routers and coordinators (at the moment the binding assumes it is configured as the coordinator). The discovery interface might be a little different and it needs some thought in how best to handle configuring the binding as a router - I would prefer to do this prior to committing to the interface.
Is this ok with you if we merge this as is?